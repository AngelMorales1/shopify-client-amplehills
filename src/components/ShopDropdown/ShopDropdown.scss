@import 'settings', 'typography';

.ShopDropdown {
  z-index: 750;
  transition: transform 0.7s ease-out;

  &--inactive {
    transform: translate3d(0, spacing(-180), 0);
  }

  &__nav-container {
    top: $header-height-desktop;
    height: spacing(70);
  }

  &__nav-container--with-alert {
    top: calc(#{$header-height-desktop} + #{$alert-height-desktop});
  }

  &__product-wrapper {
    width: 60%;
  }

  &__product-container {
    max-width: 40rem;
  }

  &__card {
    width: spacing(42);
    height: spacing(55);
    border-radius: radius('xs');
    overflow: hidden;
  }

  &__image--container {
    height: 70%;
  }

  &__image {
    object-fit: contain;
  }

  &__card-text-container {
    font-size: $font-2;
    line-height: $font-4;
    height: 20%;
  }

  &__merch-container {
    height: spacing(55);
    width: 20%;
  }
}
