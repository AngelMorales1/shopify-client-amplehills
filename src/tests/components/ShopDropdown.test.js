import React from 'react';
import { shallow } from 'enzyme';

import ShopDropdown from 'components/ShopDropdown';

it('renders without data', () => {
  const component = shallow(<ShopDropdown />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <ShopDropdown
      productLanding={{
        description: 'foo',
        hideWholesaleBlock: false,
        iceCreamDescription: '',
        iceCreamProducts: [],
        iceCreamTitle: 'foo',
        merchandiseDescription: 'foo',
        merchandiseProducts: [],
        merchandiseTitle: '',
        products: [],
        subNavigation: true,
        title: 'foo',
        wholesaleDescription: 'foo',
        wholesaleImage: {
          sys: { id: 'foo' },
          fields: { file: { url: 'foo' } }
        },
        wholesaleTitle: 'foo'
      }}
      alertIsActive={false}
      shopOnlineDropdownIsOpen={false}
      onMouseEnter={() => console.log('foo')}
      onMouseLeave={() => console.log('foo')}
    />
  );

  expect(component).toMatchSnapshot();
});
