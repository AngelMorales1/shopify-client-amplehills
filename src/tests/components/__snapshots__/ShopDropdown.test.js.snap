// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShopDropdown
    alertIsActive={false}
    closeShopDropdown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    openShopDropdown={[Function]}
    productLanding={
        Object {
            "description": "foo",
            "hideWholesaleBlock": false,
            "iceCreamDescription": "",
            "iceCreamProducts": Array [],
            "iceCreamTitle": "foo",
            "merchandiseDescription": "foo",
            "merchandiseProducts": Array [],
            "merchandiseTitle": "",
            "products": Array [],
            "subNavigation": true,
            "title": "foo",
            "wholesaleDescription": "foo",
            "wholesaleImage": Object {
              "fields": Object {
                "file": Object {
                  "url": "foo",
                },
              },
              "sys": Object {
                "id": "foo",
              },
            },
            "wholesaleTitle": "foo",
          }
    }
    shopDropdownIsOpen={false}
    shopOnlineDropdownIsOpen={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="absolute z-nav"
>
        <div
                className="w100 pt3 px4 bg-peach fixed drip"
        >
                <div
                        className="content-width mx-auto flex flex-row items-center justify-end"
                >
                        <div
                                className=""
                        >
                                <div
                                        className="flex flex-row justify-between mx-auto"
                                >
                                        
                                </div>
                        </div>
                </div>
        </div>
</div>,
      "className": "undefined relative",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 pt3 px4 bg-peach fixed drip"
>
          <div
                    className="content-width mx-auto flex flex-row items-center justify-end"
          >
                    <div
                              className=""
                    >
                              <div
                                        className="flex flex-row justify-between mx-auto"
                              >
                                        
                              </div>
                    </div>
          </div>
</div>,
        "className": "absolute z-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="content-width mx-auto flex flex-row items-center justify-end"
>
            <div
                        className=""
            >
                        <div
                                    className="flex flex-row justify-between mx-auto"
                        >
                                    
                        </div>
            </div>
</div>,
          "className": "w100 pt3 px4 bg-peach fixed drip",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className=""
>
                <div
                                className="flex flex-row justify-between mx-auto"
                >
                                
                </div>
</div>,
              null,
            ],
            "className": "content-width mx-auto flex flex-row items-center justify-end",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="flex flex-row justify-between mx-auto"
>
                  
</div>,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "flex flex-row justify-between mx-auto",
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="absolute z-nav"
>
          <div
                    className="w100 pt3 px4 bg-peach fixed drip"
          >
                    <div
                              className="content-width mx-auto flex flex-row items-center justify-end"
                    >
                              <div
                                        className=""
                              >
                                        <div
                                                  className="flex flex-row justify-between mx-auto"
                                        >
                                                  
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "className": "undefined relative",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="w100 pt3 px4 bg-peach fixed drip"
>
            <div
                        className="content-width mx-auto flex flex-row items-center justify-end"
            >
                        <div
                                    className=""
                        >
                                    <div
                                                className="flex flex-row justify-between mx-auto"
                                    >
                                                
                                    </div>
                        </div>
            </div>
</div>,
          "className": "absolute z-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="content-width mx-auto flex flex-row items-center justify-end"
>
              <div
                            className=""
              >
                            <div
                                          className="flex flex-row justify-between mx-auto"
                            >
                                          
                            </div>
              </div>
</div>,
            "className": "w100 pt3 px4 bg-peach fixed drip",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className=""
>
                  <div
                                    className="flex flex-row justify-between mx-auto"
                  >
                                    
                  </div>
</div>,
                null,
              ],
              "className": "content-width mx-auto flex flex-row items-center justify-end",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="flex flex-row justify-between mx-auto"
>
                    
</div>,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "flex flex-row justify-between mx-auto",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShopDropdown
    alertIsActive={false}
    closeShopDropdown={[Function]}
    openShopDropdown={[Function]}
    productLanding={Object {}}
    shopDropdownIsOpen={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="absolute z-nav"
>
        <div
                className="w100 pt3 px4 bg-peach fixed drip"
        >
                <div
                        className="content-width mx-auto flex flex-row items-center justify-end"
                >
                        <div
                                className=""
                        >
                                <div
                                        className="flex flex-row justify-between mx-auto"
                                >
                                        
                                </div>
                        </div>
                </div>
        </div>
</div>,
      "className": "undefined relative",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 pt3 px4 bg-peach fixed drip"
>
          <div
                    className="content-width mx-auto flex flex-row items-center justify-end"
          >
                    <div
                              className=""
                    >
                              <div
                                        className="flex flex-row justify-between mx-auto"
                              >
                                        
                              </div>
                    </div>
          </div>
</div>,
        "className": "absolute z-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="content-width mx-auto flex flex-row items-center justify-end"
>
            <div
                        className=""
            >
                        <div
                                    className="flex flex-row justify-between mx-auto"
                        >
                                    
                        </div>
            </div>
</div>,
          "className": "w100 pt3 px4 bg-peach fixed drip",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className=""
>
                <div
                                className="flex flex-row justify-between mx-auto"
                >
                                
                </div>
</div>,
              null,
            ],
            "className": "content-width mx-auto flex flex-row items-center justify-end",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="flex flex-row justify-between mx-auto"
>
                  
</div>,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "flex flex-row justify-between mx-auto",
                },
                "ref": null,
                "rendered": Array [],
                "type": "div",
              },
              "type": "div",
            },
            null,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="absolute z-nav"
>
          <div
                    className="w100 pt3 px4 bg-peach fixed drip"
          >
                    <div
                              className="content-width mx-auto flex flex-row items-center justify-end"
                    >
                              <div
                                        className=""
                              >
                                        <div
                                                  className="flex flex-row justify-between mx-auto"
                                        >
                                                  
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "className": "undefined relative",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="w100 pt3 px4 bg-peach fixed drip"
>
            <div
                        className="content-width mx-auto flex flex-row items-center justify-end"
            >
                        <div
                                    className=""
                        >
                                    <div
                                                className="flex flex-row justify-between mx-auto"
                                    >
                                                
                                    </div>
                        </div>
            </div>
</div>,
          "className": "absolute z-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="content-width mx-auto flex flex-row items-center justify-end"
>
              <div
                            className=""
              >
                            <div
                                          className="flex flex-row justify-between mx-auto"
                            >
                                          
                            </div>
              </div>
</div>,
            "className": "w100 pt3 px4 bg-peach fixed drip",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className=""
>
                  <div
                                    className="flex flex-row justify-between mx-auto"
                  >
                                    
                  </div>
</div>,
                null,
              ],
              "className": "content-width mx-auto flex flex-row items-center justify-end",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="flex flex-row justify-between mx-auto"
>
                    
</div>,
                  "className": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                    "className": "flex flex-row justify-between mx-auto",
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "div",
                },
                "type": "div",
              },
              null,
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
